Sources/MapboxCoreNavigation/Dictionary.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+extension Dictionary where Value == Any {
+    mutating func deepMerge(with dictionary: Dictionary) {
```

Comment:
It isn't clear which value will be used during the merge(a new one or current). Maybe it's better to provide a closure here just like a standard `merge` does?